File: StewardApi/Contracts/Common/MasterInventoryItem.cs
using System;

namespace Turn10.LiveOps.StewardApi.Contracts.Common
{
    /// <summary>
    ///     Represents a master inventory item.
    /// </summary>
    public class MasterInventoryItem
    {
        /// <summary>
        ///     Gets or sets the item ID.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        ///     Gets or sets the item description.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        ///     Gets or sets the item quantity.
        /// </summary>
        public int Quantity { get; set; }

        /// <summary>
        ///     Gets or sets the item error.
        /// </summary>
        public object Error { get; set; }
    }
}

File: packages/StewardClient/src/app/shared/models/master-inventory-item.ts
import BigNumber from 'bignumber.js';
import { ApolloMasterInventory } from './apollo';
import { MSError } from './error.model';
import { SunriseMasterInventory } from './sunrise/sunrise-master-inventory.model';

/** Interface for a master inventory item. */
export interface MasterInventoryItem {
  id: BigNumber;
  description: string;
  quantity: number;
  itemType: string;
  error: MSError;
}

export type MasterInventoryUnion = SunriseMasterInventory | ApolloMasterInventory;
